### DISCUSSION TITLE
hourly Preceding hour values

### DISCUSSION THREAD

**Discussion by farfromrefug** [open-meteo/open-meteo#317] [Q&A] [answered]
Hi,
I dont understand the meaning of preceding hour values. If i take for example `precipitation` the value represents the inches of precipitation of the hour before? Which means that in an app showing hourly weather forecast i need to look at the `hourly.precipitation[index-1]` ?
If so it seems really weird, why not returning the values at the correct hour? It also means that i dont have precipitation actual value for index 0.
May be i misunderstood the doc
Thanks for your amazing work

**Comment by patrick-zippenfenig** [comment, marked as answer]
Hi,
many weather variables are indicated at the exact hour and 0 minutes. E.g. Temperature at 12:00 is the temperature at exactly this instant timestep. For precipitation and other fluxes like solar radiation this does not work well as the precipitation rate at exactly this moment in time does not contain much information. 

Similar to weather stations, the sum over the previous 60 minutes is used. Precipitation at 12:00 is the sum of all rain from 11:00-12:00. This definition is standard for weather data and used in other APIs as well. In many weather apps and websites the difference between instant values and "preceding hour sums" is ignored because forecast accuracy is a more limiting factor.

If you want to be precise, indicate that the precipitation if given for this one hour interval.

In the future there will also be more datasets with 15-minutely data. For example the German weather service offers 15minutes precipitation and solar radiations for central Europe: https://open-meteo.com/en/docs/dwd-api

**Reply by farfromrefug** [reply]
@patrick-zippenfenig thanks for the great answer. ANd i now realise that OWM mention that those values are from the last hour! 
I think i need to show the index-1 value for precipitation and other values. Most people use hourly as an indicitation like "will it rain between 4pm and 5pm". Right now with owm and om i show them data which by comparaison to what they expect is false.
Thanks again!
